app/openvpn_client_lwt.ml

```diff
@@ -62,20 +62,42 @@ let rec write_to_fd fd data =
          Lwt_result.lift
            (Rresult.R.error_msgf "write error %s" (Printexc.to_string e)))
 
+let write_udp sa fd data =
+  Lwt.catch (fun () ->
+      Lwt_unix.sendto fd (Cstruct.to_bytes data) 0 (Cstruct.len data) [] sa >|= fun _ ->
+      Ok ())
```

Comment:
errors are handled by lwt_unix, and result in an exception. the only other error can be a short write, and it is not clear to me whether "send it again" is a good recovery strategy (since udp is packet-oriented, we can't just send the "missing bytes"). I added a log message.

Reply:
I think that short writes cannot happen with plain vanilla UDP, I think it sends either the whole datagram or nothing. Maybe we can use `sendto` for both TCP and UDP though, in which case we would need to check the length. Anyway, thanks for the logg message! :)